### PR TITLE
boards: model nrf52 boards

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Follow up to https://github.com/RIOT-OS/RIOT/pull/16837 this models the remaining BOARDs
I had to do multiple changes to model this one, so commits are better checked one by one.

### Testing procedure

- Gren CI
- Check that driver modeling still makes sense
- Check that default bootloader config makes sense

### Issues/PRs references

part of #16875

### CODE REVIEW

**Comment by fjmolinas** [comment]
I had quite some trouble with this one, the common nrf52 dependencies are a complete mess: some select the common boards/nrf* something dependencies, but not the features, or the dependencies but not the common module, or the features and not the common module. Anyway, seems like its finally right, can I squash @leandrolanzieri to clean this up a bit?

**Comment by leandrolanzieri** [comment]
> I had quite some trouble with this one, the common nrf52 dependencies are a complete mess: some select the common boards/nrf* something dependencies, but not the features, or the dependencies but not the common module, or the features and not the common module. Anyway, seems like its finally right, can I squash @leandrolanzieri to clean this up a bit?

Sure, go ahead. It's gonna be easier to review

**Comment by fjmolinas** [comment]
Squashed

File: cpu/nrf52/periph/Kconfig [hunk 1]
```diff
@@ -19,5 +19,13 @@ config MODULE_PERIPH_SPI
 config MODULE_SAUL_NRF_VDDH
     bool "Internal Voltage Sensor"
     depends on HAS_PERIPH_ADC
+    select MODULE_PERIPH_ADC
 
 endif # TEST_KCONFIG
+
+config HAVE_SAUL_NRF_VDDH
+    bool
+    select MODULE_SAUL_NRF_VDDH if MODULE_SAUL_DEFAULT && TEST_KCONFIG
```
Comment (leandrolanzieri) [inline cpu/nrf52/periph/Kconfig:28]: `&& TEST_KCONFIG` is not required, as `MODULE_SAUL_DEFAUL` already depends on it.

**Review by leandrolanzieri — COMMENTED** [review]
Looks good to me already. Just some minor nits. Let's see what the CI thinks

**Comment by fjmolinas** [comment]
@leandrolanzieri can I squash if green and remove the .murdock change?

**Comment by leandrolanzieri** [comment]
> @leandrolanzieri can I squash if green and remove the .murdock change?

Go ahead

**Comment by fjmolinas** [comment]
Damn, recent changes broke this PR again... :( will adapt.

**Comment by fjmolinas** [comment]
> Damn, recent changes broke this PR again... :( will adapt.

Changes upstream

**Comment by fjmolinas** [comment]
All green will squash.

**Comment by leandrolanzieri** [comment]
@fjmolinas you still need to remove 03bb3a6556b4d21eec1a11f6679e2a1ade709a90 and pick one board for testing.

**Comment by leandrolanzieri** [comment]
Also 10a5d96456c5155cedd58e9f9dadad7dfe739d25 seems unrelated

**Comment by fjmolinas** [comment]
> @fjmolinas you still need to remove [03bb3a6](https://github.com/RIOT-OS/RIOT/commit/03bb3a6556b4d21eec1a11f6679e2a1ade709a90) and pick one board for testing.

There are already nrf52 boards listed.

File: drivers/lpsxxx/Kconfig [hunk 2]
```diff
@@ -6,38 +6,42 @@
...
+        (LPS331AP/LPS25HB/LPS22HB/LPS22HH). Select a model.
+
+if MODULE_LPSXXX
+
+choice LPSXXX_VARIANT
+    bool "Model"
+    default MODULE_LPS331AP if HAVE_LPS331AP
+    default MODULE_LPS22HB if HAVE_LPS22HB
+    default MODULE_LPS22HH if HAVE_LPS22HH
+    default MODULE_LPS25HB if HAVE_LPS25Hb
```
Comment (leandrolanzieri) [inline drivers/lpsxxx/Kconfig:26]: Sorry, just spotted this one
```suggestion
    default MODULE_LPS25HB if HAVE_LPS25HB
```

**Review by leandrolanzieri — COMMENTED** [review]
Sorry I just spotted this now. Also there is a typo in the message of f54fdf30cb43850fb167442f18c2f7566976f21b

**Comment by leandrolanzieri** [comment]
Feel free to squash directly

**Review by leandrolanzieri — APPROVED** [review]
ACK, this was tested on the CI over all nrf52 boards.

**Pull request merged**